pdm/installers/_editable_install.py

```diff
@@ -23,7 +23,7 @@ def install(setup_py, prefix, lib_dir, bin_dir):
         "--site-dirs={0}".format(lib_dir),
     ]
     sys.path.append(lib_dir)
-    exec(compile(code, __file__, "exec"))
+    exec(compile(code, __file__, "exec"), globals())
```

Comment:
```suggestion
    exec(compile(code, __file__, "exec"), {})
```
I don't want globals() to contaminate the execution globals. An empty dict just does the trick from my testing result.

Reply:
It does not fix the referenced issue, where it still fails with a `NameError` for the `import re` from the toplevel of mkdoc's setup.py.

I think it would be good to have a test for this already, which would show the issue / verify the fix, but I have not looked into the existing tests, and how it fits best yet.

Reply:
I am curious in what case the script requires `globals()`, since I tried:

```python
text = 'import re\ndef foo(): print(re)\nfoo()\n'
exec(compile(text, 'main', 'exec'))
```
And it works smoothly, also, as I mentioned earlier, this snippet is actually being used by `pip` to execute all `setup.py` commands

Reply:
Got it, I should move the code out of the `install()` function